runtime/lua/vim/lsp/util.lua

```diff
@@ -1353,6 +1353,39 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+-- for use with `formatexpr` if called without parms
+-- @param start_line 1-indexed line
+-- @param end_line 1-indexed line
+-- @param timeout_ms optional
+function M.formatexpr(start_line, end_line, timeout_ms)
+  if not start_line or not end_line then
+    if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
+      -- `formatexpr` is also called when exceding
+      -- `textwidth` in insert mode
+      -- fall back to internal formatting
+      return 1
+    end
+    start_line = vim.v.lnum
```

Comment:
Should this be `start_line = start_line or vim.v.lnum` so that if you pass in `start_line` but not `end_line` then it respects it? Or is it that you have to pass in both or nothing? Not sure which way would be the best.

Reply:
It dosen't make much sense to use pass only `start_line` and using `v:count` for `end_line` as that is `The count given for the last Normal mode command.` 
I guess it could be set the last line in the buffer if you only pass in a start.